File: cs50/PSET2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // get input text

    string text = get_string("Text: ");

    float l = 0;
    float w = 1;
    float s = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // calculate the number of letters

        if (isalpha(text[i]) != 0)
        {
            l++;
        }

        // calculate the number of words

        if (text[i] == 32)
        {
            w++;
        }

        // calculate the number sentences

        if (text[i] == 46 || text[i] == 63 || text[i] == 33)
        {
            s++;
        }
    }

    // calculate the Coleman-Liau index

    float L = 100 * l / w;
    float S = 100 * s / w;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // print the results based on index

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: cs50/PSET6/sentimental-readability/readability.py
# import libraries
import string
from cs50 import get_string

text = get_string("Text: ")

# count letters
letters = 0

for i in text:
    if (i.isalpha()):
        letters += 1

# count words
words = len(text.split())

# count sentences
sentences = 0

for j in text:
    if (j == '.' or j == '?' or j == '!'):
        sentences += 1

# calculate index
x = words / 100
l = letters / x
s = sentences / x

index = round(0.0588 * l - 0.296 * s - 15.8)

# print grades
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
